src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_quantization.cpp

```diff
@@ -365,6 +365,16 @@ void prepare_quantization::prepare_dequantize_merge(program& p, eltwise_node& el
                     break;
                 }
             }
+
+            // Check if eltwise_dep and eltwise_node have same output layout
+            if (eltwise_dep.get_outputs_count() > 0 && eltwise_node.get_outputs_count() > 0) {
```

Comment:
Are the number of elements in mem0 and mem1 equal? Can we compare only constants' layouts instead of the output layouts?